backend/src/apiserver/server/util.go

```diff
@@ -47,6 +47,14 @@ func GetPipelineName(queryString string, fileName string) (string, error) {
 	return pipelineName, nil
 }
 
+func GetPipelineNamespace(queryString string) (string, error) {
+	pipelineNamespace, err := url.QueryUnescape(queryString)
+	if err != nil {
+		return "", util.NewInvalidInputErrorWithDetails(err, "Pipeline namespace in the query string has invalid format.")
+	}
+	return pipelineNamespace, nil
+}
+
```

Comment:
nit: Since this function is only used in the upload server, maybe keep it there?

Reply:
Makes sense. Done.